src/enums.rs

```diff
@@ -380,7 +484,39 @@ pub enum Launchable {
     Unknown(String),
 }
 
-#[derive(Clone, Debug, Serialize, PartialEq)]
+impl Serialize for Launchable {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut s = serializer.serialize_struct("launchable", 2)?;
+        match self {
+            Launchable::DesktopId(url) => {
+                s.serialize_field("type", "desktop_id")?;
+                s.serialize_field("name", &url)?;
+            }
+            Launchable::Service(url) => {
+                s.serialize_field("type", "service")?;
+                s.serialize_field("name", &url)?;
+            }
+            Launchable::Url(url) => {
+                s.serialize_field("type", "url")?;
+                s.serialize_field("name", &url)?;
+            }
+            Launchable::CockpitManifest(url) => {
+                s.serialize_field("type", "cockpit_manifest")?;
+                s.serialize_field("name", &url)?;
+            }
+            Launchable::Unknown(url) => {
```

Comment:
I would say, just ignore that url?

Reply:
I don't think that would be a good idea. We should serialize the whole `Component` struct completely and don't ignore stuff.